stixcore/tmtc/packets.py

```diff
@@ -0,0 +1,317 @@
+
+    def __init__(self, data):
+        """
+
+        Parameters
+        ----------
+        data
+        """
+        if not isinstance(data, TMPacket):
+            self.source_packet_header = SourcePacketHeader(data)
+            self.data_header = TMDataHeader(self.source_packet_header.bitstream)
+        else:
+            self.source_packet_header = data.source_packet_header
+            self.data_header = data.data_header
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.source_packet_header}, {self.data_header})'
+
+    def __str__(self):
+        return self.__repr__()
+
+
+class TM_1_1(GenericTMPacket):
+    """
+    TM(1,1) Telecommand acceptance report
+    """
+    def __init__(self, data):
+        super().__init__(data)
+        structure = {'NIX00001': 'uint:16', 'NIX00002': 'uint:16'}
+        res = parse_bitstream(self.source_packet_header.bitstream, structure)
+        [setattr(self, key, value) for key, value in res['fields'].items()]
+
+    @classmethod
+    def is_datasource_for(cls, tm_packet):
+        dh = tm_packet.data_header
+        return dh.service_type == 1 and dh.service_subtype == 1
+
+
+class TM_21_6_30(GenericTMPacket):
+    """
+    TM(21, 6) SSID 30
+    """
+    def __init__(self, data):
+        super().__init__(data)
+        attr_dict = dict()
+        fixed_structure = {
+            'NIX00120': 'uint:8',
+            'NIX00445': 'uint:32',
+            'NIX00446': 'uint:16',
+            'NIX00405': 'uint:16',
+            'NIX00407': 'uint:32',
+            'spare1': 'uint:4',
+            'NIXD0407': 'uint:12',
+            'spare2': 'uint:1',
+            'NIXD0101': 'uint:1',
+            'NIXD0102': 'uint:3',
+            'NIXD0103': 'uint:3',
+            'NIXD0104': 'uint:1',
+            'NIXD0105': 'uint:3',
+            'NIXD0106': 'uint:3',
+            'NIXD0107': 'uint:1',
+            'NIX00266': 'uint:32',
+            'NIX00270': 'uint:8'
+        }
+        fixed = parse_bitstream(self.source_packet_header.bitstream, fixed_structure)
+        attr_dict.update(fixed['fields'])
+        num_energies = fixed['fields']['NIX00270']
+        # Peak ahead but don't advance pos index
+        num_data_points = self.source_packet_header.bitstream.peek('uint:16')
+        repeaters_structures = [({'NIX00271': ', '.join(['uint:16']),
+                                  'NIX00272': ', '.join(['uint:8']*num_data_points)}, num_energies),
+                                ({'NIX00273': 'uint:16'}, 1),
+                                ({'NIX00274': ', '.join(['uint:8']*num_data_points)}, 1),
+                                ({'NIX00275': 'uint:16'}, 1),
+                                ({'NIX00276': ', '.join(['uint:8']*num_data_points)}, 1)]
+
+        for i, (structure, repeats) in enumerate(repeaters_structures):
+            dynamic = parse_repeated(self.source_packet_header.bitstream, structure, repeats)
+            attr_dict.update(dynamic['fields'])
+
```

Comment:
@drhlxiao same kind of question I've hard coded this example but ideally would like to be able to query the IDB using your routines to obtain the fixed and dynamic structure based on the service type, subtype and SSID

Reply:
@nicHoch I think it would be good for you to review this part and we can merge and then I can review the other part in #19. What do you think?